### PR TITLE
Update CONTRIBUTING.md to expose the requirements to open a PR. Closes #224

### LINKED ISSUE #224: PR requirements not listed in CONTRIBUTING.md
**Description**

Hello! I'm making my first contribution to the repo and when I was about to create the PR I faced the following requirements:
```
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!
```
But the thing is that those requirements were not listed on the CONTRIBUTING.md, so some people (including me haha) could end up having to redo some work.

I think a good idea would be update the CONTRIBUTING.md text to add those requirements.
If you agree, I would be happy to raise a PR with that =)

### PR DESCRIPTION
**Pull request by diogoteles08**
Signed-off-by: Diogo Teles Sant'Anna <diogoteles08@gmail.com>

#### Summary

This PR was created because the Template for creating a PR have a text that says it's necessary to have all commits signed-off to make the PR, but it was not said on the CONTRIBUTING.md.

IMPORTANT: on the PR template it's still present the item "lastly, ensure there are no merge commits!". If this step is not necessary anymore, it would be better to have it removed. I could not find where to change this text, so I supposed it was configured directly on github. Let me know if I can help with this as well.

### CODE REVIEW

**Comment by diogoteles08** [comment]
So, I have taken the liberty to try to add a specific PR template to the repo as described [here](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository), to see if it would overwrite the current template, wherever it comes from haha
And testing on my fork repo, it seemed to work. So you can use it to describe the specific PR template for this repo, and it's a bit more transparent and "changeable" for the future =)
Does it makes sense to you?

**Comment by woodruffw** [comment]
Looks fine to me. This should indeed override the org-level template; I've done it with other orgs in the past.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -139,3 +139,6 @@ bug reports when their debug logs include helpful context!
 * *Update the [CHANGELOG](./CHANGELOG.md)*. If your changes are public or result
 in changes to `sigstore`'s CLI, please record them under the "Unreleased" section,
 with an entry in an appropriate subsection ("Added", "Changed", "Removed", or "Fixed").
+
+* Ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin).
```
Comment (woodruffw) [inline CONTRIBUTING.md:143]: Just a small grammar nit + formatting nit:

```suggestion
* Ensure your commits are signed off, as sigstore uses the
[DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin). 
```

**Review by woodruffw — APPROVED** [review]

**Pull request merged**